src/libpmem2/config.c

```diff
@@ -86,3 +87,31 @@ pmem2_config_delete(struct pmem2_config **cfg)
 	*cfg = NULL;
 	return 0;
 }
+
+/*
+ * pmem2_config_set_required_store_granularity -- set granularity
+ * requested by user in the pmem2_config structure
+ */
+int
+pmem2_config_set_required_store_granularity(struct pmem2_config *cfg,
+		enum pmem2_granularity g)
+{
+	if (!cfg) {
```

Comment:
We don't check this anywhere and I think we shouldn't. If you really want to, it should be introduced as a separate PR.